### PR TITLE
Add validation on the table route + Add feature to update users

### PR DESCRIPTION
**Pull request by AbegaM**
### Modifications 
1. Added a feature that throws an error when a user tries to create a `reserved` table name
2. Added a feature that updates users data 
    *  when users send a request to the `PUT /api/tables/_users/rows` endpoint, the `processRowRequest` function will remove some fields like `salt`, `hashed_password` and `is_superuser` from the request body

### CODE REVIEW

File: src/constants/tables.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
```
Comment (IanMayo) [inline src/constants/tables.js:1]: Thanks @AbegaM 

I'd really like this list of strings to be declared as `const` constants, and referred to in all instances of use.  That will avoid the potential for typos such as `users` for `_users`.

File: src/constants/tables.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  reservedTableNames: [
```
Comment (thevahidal) [inline src/constants/tables.js:2]: I think it's better to have these as an object, e.g.
```
{
    _USERS: "_users",
    ...
}
```

File: src/controllers/auth.js [hunk 3]
```diff
@@ -24,7 +25,7 @@ const createDefaultTables = async () => {
   // create _users table
   if (!usersTable) {
     // create the _users table
-    tableService.createTable('_users', dbTables.userSchema);
+    tableService.createTable('_users', schema.userSchema);
```
Comment (thevahidal) [inline src/controllers/auth.js:28]: Then we can use it here, e.g.
```
tableService.createTable(reservedTableNames._USERS, schema.userSchema);
```

File: src/middlewares/api.js [hunk 4]
```diff
@@ -13,40 +18,61 @@ const processRequest = async (req, res, next) => {
...
+    // check if the username is taken
+    if (fields.username) {
+      if (isUsernameTaken(fields.username)) {
+        return res
+          .status(409)
+          .send({ message: 'This username is already taken' });
+      }
+    }
+
+    // remove some user fields from the request like (is_superuser, hashed_password, salt). NOTE: password can be updated via the /change-password API and superuser status can be only updated from the CLI
```
Comment (thevahidal) [inline src/middlewares/api.js:37]: For this kind of long comments, it's better to use Multi-line Comments

**Review by thevahidal — CHANGES_REQUESTED** [review]

File: src/constants/tables.js (hunk 2 above)
Comment (AbegaM) [reply src/constants/tables.js:2]: This comment is fixed in the next PR https://github.com/thevahidal/soul/pull/160

File: src/controllers/auth.js (hunk 3 above)
Comment (AbegaM) [reply src/controllers/auth.js:28]: This comment is fixed in the next PR https://github.com/thevahidal/soul/pull/160

**Review by thevahidal — APPROVED** [review]

**Pull request merged**